src/Compilers/CSharp/Portable/FlowAnalysis/ExitPointsWalker.cs

```diff
@@ -41,27 +37,28 @@ protected override void Free()
             base.Free();
         }
 
-        internal static IEnumerable<StatementSyntax> Analyze(CSharpCompilation compilation, Symbol member, BoundNode node, BoundNode firstInRegion, BoundNode lastInRegion)
+        internal static ImmutableArray<SyntaxNode> Analyze(CSharpCompilation compilation, Symbol member, BoundNode node, BoundNode firstInRegion, BoundNode lastInRegion)
         {
             var walker = new ExitPointsWalker(compilation, member, node, firstInRegion, lastInRegion);
             try
             {
                 bool badRegion = false;
-                walker.Analyze(ref badRegion);
-                var result = walker._branchesOutOf.ToImmutableAndFree();
-                walker._branchesOutOf = null;
-                return badRegion ? SpecializedCollections.EmptyEnumerable<StatementSyntax>() : result;
+                var result = walker.Analyze(ref badRegion);
+                return badRegion ? ImmutableArray<SyntaxNode>.Empty : result;
```

Comment:
>return badRegion ? ImmutableArray<SyntaxNode>.Empty : result; [](http://example.com/codeflow?start=16&length=61)

Would it make sense to move this logic to the instance Analyze too, and avoid sorting + array allocation for bad region case?

Reply:
It looks like we will be able to remove the out parameter then.